File: rust/src/lc_arrays_hashing/lc_0217_contains_duplicate.rs
pub struct Solution;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        use std::collections::HashSet;

        let mut set = HashSet::new();
        for n in nums {
            if !set.insert(n) {
                return true;
            }
        }

        false
    }
}

#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn ex1() {
        assert!(Solution::contains_duplicate(vec![
            1, 2, 3, 1
        ]))
    }
}

File: python/src/lc_arrays_hashing/lc_0217_contains_duplicate.py
from typing import List

# O(n) time
# O(n) space
def contains_duplicate(nums: List[int]) -> bool:
    hashset = set()

    for n in nums:
        if n in hashset:
            return True
        hashset.add(n)
    return False

#Tests
def test_ex1():
    result = contains_duplicate([1, 2, 3, 1])
    assert result == True